add some info for adding new quantity in obs_kind

myying opened the pull request:
I was adding a new observation type in DART following the instructions in this page. I missed the note on adding new quantity definitions in obs_kind, which takes me some time to figure out. I added more information on creating new quantities, it will be easier for readers to follow.

timhoar commented on docs/pages/Getting_Started.md:
```diff
@@ -2306,10 +2307,19 @@ as shown in the last line this need not be the case. The third column must be
+
+<span id="newQuantity" class="anchor"></span> [](#newQuantity)
+### Obs is a new quantity
+To add a new quantity that is not yet defined in DART, the
```
'To add a new quantity that is not yet defined in DART' ... is redundant.  It cannot be a new quantity if it is already defined in DART.

timhoar replied:
Michael,

There is more to it than that.  You have to not only add the example as you show, you have to add another line way later in the file. The whole process is described in the obs_def_mod.html ... I just have to figure out how to link to it in stupid github-land.

timhoar replied:
GitHub does not render html ... and it only renders certain .md files.
GitHub takes the docs/pages/*md files and creates GitHub Pages html files, but if you already have html ... you're sunk.

myying replied:
> Michael,
> 
> There is more to it than that. You have to not only add the example as you show, you have to add another line way later in the file. The whole process is described in the obs_def_mod.html ... I just have to figure out how to link to it in stupid github-land.

Hey Tim,
You are right, I completely forgot the later initialization line like this
obs_kind_names(367) = obs_kind_type(QTY_ABSOLUTE_HUMIDITY, 'QTY_ABSOLUTE_HUMIDITY')
Shall I close this pull request and open a new one with this additional line, or can this pull request be somehow modified?

Also, have you tried one of those html to markdown converters online? Just a thought.

Michael

timhoar commented:
I have tried the converters. The results still require a lot of
hand-editing, and the markdown renders differently locally than on GitHub.
We have 212 html documents ... granted - some of them can wait.  The
long-term goal is to replace the .html with in-line documentation generated
by FORD.
Long-term.

As for the pull request - I like the link to the section, but I'm having
trouble figuring out the wording for the link.
'Obs is a new quantity'    isn't really enough ... Maybe 'Supporting new
observations' ... something like that.

I'd prefer to work with you on the PR rather than torture you with my
ill-posed comments.

Tim Hoar
Data Assimilation Research Section
National Center for Atmospheric Research
thoar@ucar.edu
303.497.1708


On Tue, Mar 10, 2020 at 11:12 PM Yue (Michael) Ying <
notifications@github.com> wrote:

> *@myying* commented on this pull request.
> ------------------------------
>
> In docs/pages/Getting_Started.md
> <https://github.com/NCAR/DART/pull/7#discussion_r390749174>:
>
> > @@ -2306,10 +2307,19 @@ as shown in the last line this need not be the case. The third column must be
>  the keyword COMMON_CODE which tells the preprocess program to automatically
>  generate all necessary interface code for this type. COMMON_CODE also implicitly
>  means the "model_interpolate()" routine is used as the forward operator.
> -
> -:dart: *NOTE*: to add a new quantity, the
> +
> +<span id="newQuantity" class="anchor"></span> [](#newQuantity)
> +### Obs is a new quantity
> +To add a new quantity that is not yet defined in DART, the
>
> Michael,
>
> There is more to it than that. You have to not only add the example as you
> show, you have to add another line way later in the file. The whole process
> is described in the obs_def_mod.html ... I just have to figure out how to
> link to it in stupid github-land.
>
> Hey Tim,
> You are right, I completely forgot the later initialization line like this
> obs_kind_names(367) = obs_kind_type(QTY_ABSOLUTE_HUMIDITY,
> 'QTY_ABSOLUTE_HUMIDITY')
> Shall I close this pull request and open a new one with this additional
> line, or can this pull request be somehow modified?
>
> Also, have you tried one of those html to markdown converters online? Just
> a thought.
>
> Michael
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/DART/pull/7?email_source=notifications&email_token=ADTRYOTFMJDMIDOSGBJAKBLRG4MSHA5CNFSM4LFFSZWKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCYZZLGI#discussion_r390749174>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADTRYOSUSTM4CJTLGMMWQ4TRG4MSHANCNFSM4LFFSZWA>
> .
>

myying commented:
Will it work better if I move the "new quantity" part into the "complexObs" section? Because adding a new quantity is actually one step in creating a complex observation needing special handling.

timhoar commented:
Even 'simple' observation quantities may need to be added, so I don't think
it necessarily fits into 'complex' ...

Tim Hoar
Data Assimilation Research Section
National Center for Atmospheric Research
thoar@ucar.edu
303.497.1708


On Wed, Mar 11, 2020 at 10:35 AM Yue (Michael) Ying <
notifications@github.com> wrote:

> Will it work better if I move the "new quantity" part into the
> "complexObs" section? Because adding a new quantity is actually one step in
> creating a complex observation needing special handling.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/DART/pull/7#issuecomment-597737416>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADTRYOWBVXRXUPFUNZVXRZLRG64T3ANCNFSM4LFFSZWA>
> .
>

timhoar commented:
Michael -- I worked up something that I think addresses your issues ... let
me know if you like what I've done with

https://github.com/timhoar/DART/blob/getting_started_page/docs/pages/Getting_Started.md

And did you get a notice that I issued a pull request (#8) to   NCAR/DART ?
You are not 'watching' it, so I suspect you did not get a notice.

Tim Hoar
Data Assimilation Research Section
National Center for Atmospheric Research
thoar@ucar.edu
303.497.1708


On Wed, Mar 11, 2020 at 11:38 AM Tim Hoar <thoar@ucar.edu> wrote:

> Even 'simple' observation quantities may need to be added, so I don't
> think it necessarily fits into 'complex' ...
>
> Tim Hoar
> Data Assimilation Research Section
> National Center for Atmospheric Research
> thoar@ucar.edu
> 303.497.1708
>
>
> On Wed, Mar 11, 2020 at 10:35 AM Yue (Michael) Ying <
> notifications@github.com> wrote:
>
>> Will it work better if I move the "new quantity" part into the
>> "complexObs" section? Because adding a new quantity is actually one step in
>> creating a complex observation needing special handling.
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NCAR/DART/pull/7#issuecomment-597737416>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ADTRYOWBVXRXUPFUNZVXRZLRG64T3ANCNFSM4LFFSZWA>
>> .
>>
>

myying commented:
Hey Tim,

I like your edits a lot, it makes more sense. Thanks! I'm closing my old
pull request.

I am now watching the NCAR/DART repository, I though forking it was enough,
apparently not. Now I shall have email notices of changes.

Best,
Michael

On Fri, Mar 13, 2020 at 10:19 AM Tim Hoar <notifications@github.com> wrote:

> Michael -- I worked up something that I think addresses your issues ... let
> me know if you like what I've done with
>
>
> https://github.com/timhoar/DART/blob/getting_started_page/docs/pages/Getting_Started.md
>
> And did you get a notice that I issued a pull request (#8) to NCAR/DART ?
> You are not 'watching' it, so I suspect you did not get a notice.
>
> Tim Hoar
> Data Assimilation Research Section
> National Center for Atmospheric Research
> thoar@ucar.edu
> 303.497.1708
>
>
> On Wed, Mar 11, 2020 at 11:38 AM Tim Hoar <thoar@ucar.edu> wrote:
>
> > Even 'simple' observation quantities may need to be added, so I don't
> > think it necessarily fits into 'complex' ...
> >
> > Tim Hoar
> > Data Assimilation Research Section
> > National Center for Atmospheric Research
> > thoar@ucar.edu
> > 303.497.1708
> >
> >
> > On Wed, Mar 11, 2020 at 10:35 AM Yue (Michael) Ying <
> > notifications@github.com> wrote:
> >
> >> Will it work better if I move the "new quantity" part into the
> >> "complexObs" section? Because adding a new quantity is actually one
> step in
> >> creating a complex observation needing special handling.
> >>
> >> —
> >> You are receiving this because you commented.
> >> Reply to this email directly, view it on GitHub
> >> <https://github.com/NCAR/DART/pull/7#issuecomment-597737416>, or
> >> unsubscribe
> >> <
> https://github.com/notifications/unsubscribe-auth/ADTRYOWBVXRXUPFUNZVXRZLRG64T3ANCNFSM4LFFSZWA
> >
> >> .
> >>
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/DART/pull/7#issuecomment-598803028>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB5KVUDKTLJQIZR3ZFJENI3RHJMHNANCNFSM4LFFSZWA>
> .
>


-- 
Yue (Michael) Ying
Postdoctoral Fellow
Advanced Study Program, NCAR, Boulder, Colorado
Tel: (303)-497-2497

timhoar commented:
Thanks Michael -   our review process is such that I should not accept my
own PR, so I've asked Jeff Steward to review it.


Tim Hoar
Data Assimilation Research Section
National Center for Atmospheric Research
thoar@ucar.edu
303.497.1708


On Sat, Mar 14, 2020 at 10:35 AM Yue (Michael) Ying <
notifications@github.com> wrote:

> Hey Tim,
>
> I like your edits a lot, it makes more sense. Thanks! I'm closing my old
> pull request.
>
> I am now watching the NCAR/DART repository, I though forking it was enough,
> apparently not. Now I shall have email notices of changes.
>
> Best,
> Michael
>
> On Fri, Mar 13, 2020 at 10:19 AM Tim Hoar <notifications@github.com>
> wrote:
>
> > Michael -- I worked up something that I think addresses your issues ...
> let
> > me know if you like what I've done with
> >
> >
> >
> https://github.com/timhoar/DART/blob/getting_started_page/docs/pages/Getting_Started.md
> >
> > And did you get a notice that I issued a pull request (#8) to NCAR/DART ?
> > You are not 'watching' it, so I suspect you did not get a notice.
> >
> > Tim Hoar
> > Data Assimilation Research Section
> > National Center for Atmospheric Research
> > thoar@ucar.edu
> > 303.497.1708
> >
> >
> > On Wed, Mar 11, 2020 at 11:38 AM Tim Hoar <thoar@ucar.edu> wrote:
> >
> > > Even 'simple' observation quantities may need to be added, so I don't
> > > think it necessarily fits into 'complex' ...
> > >
> > > Tim Hoar
> > > Data Assimilation Research Section
> > > National Center for Atmospheric Research
> > > thoar@ucar.edu
> > > 303.497.1708
> > >
> > >
> > > On Wed, Mar 11, 2020 at 10:35 AM Yue (Michael) Ying <
> > > notifications@github.com> wrote:
> > >
> > >> Will it work better if I move the "new quantity" part into the
> > >> "complexObs" section? Because adding a new quantity is actually one
> > step in
> > >> creating a complex observation needing special handling.
> > >>
> > >> —
> > >> You are receiving this because you commented.
> > >> Reply to this email directly, view it on GitHub
> > >> <https://github.com/NCAR/DART/pull/7#issuecomment-597737416>, or
> > >> unsubscribe
> > >> <
> >
> https://github.com/notifications/unsubscribe-auth/ADTRYOWBVXRXUPFUNZVXRZLRG64T3ANCNFSM4LFFSZWA
> > >
> > >> .
> > >>
> > >
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/NCAR/DART/pull/7#issuecomment-598803028>, or
> > unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AB5KVUDKTLJQIZR3ZFJENI3RHJMHNANCNFSM4LFFSZWA
> >
> > .
> >
>
>
> --
> Yue (Michael) Ying
> Postdoctoral Fellow
> Advanced Study Program, NCAR, Boulder, Colorado
> Tel: (303)-497-2497
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/DART/pull/7#issuecomment-599093376>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADTRYOSQKCRKYFDLMHRYFKDRHOW63ANCNFSM4LFFSZWA>
> .
>
